src/pybind/mgr/dashboard/services/orchestrator.py

```diff
@@ -170,6 +170,36 @@ def action(self, daemon_name='', action='', image=None):
         return self.api.daemon_action(daemon_name=daemon_name, action=action, image=image)
 
 
+class UpgradeManger(ResourceManager):
+    @wait_api_result
+    def list(self, image: Optional[str], tags: bool,
+             show_all_versions: Optional[bool]) -> Dict[Any, Any]:
+        return self.api.upgrade_ls(image, tags, show_all_versions)
+
+    @wait_api_result
+    def status(self):
+        return self.api.upgrade_status()
+
+    @wait_api_result
+    def start(self, image: str, version: str, daemon_types: Optional[List[str]] = None,
```

Comment:
these should be `Optional` fields. And let's not expose the other fields like `daemon_types`, `host_placement`, `services`, `limit` right now. We can deal with it later.
```suggestion
    def start(self, image: Optional[str], version: Optional[str], daemon_types: Optional[List[str]] = None,
```

Reply:
Ok I see they're optional in orch module as well. Well those additional fields are anyways Optional so we could have them expose right now so later once we have requirement for them we could use it, because it's just dashboard UI will not have workflow based on daemon_types but from APIs I believe we could still do upgrade will all options as well. Wdyt?